website/src/pages/fonts/[font].tsx

```diff
@@ -58,19 +75,54 @@ export default function FontPage({ metadata, defPreviewText }: FontPageProps) {
 
 export const getStaticProps: GetStaticProps = async ({ params }) => {
   try {
-    const metadata: MetadataProps = await fetcher(
+    const metadata: MetadataProps = await fetchJson(
       fontsourceDownload.data(`${params.font}`).metadata
     );
 
+    if (!metadata) return { notFound: true };
+
     // Generates preview texts in matching languages
     const defPreviewText = selectDefPreviewText(
       metadata.fontId,
       metadata.defSubset
     );
 
-    return metadata
-      ? { props: { metadata, defPreviewText }, revalidate: 7200 }
-      : { notFound: true };
+    const fontCssNames: [number, string[]][] = [];
+    // Fetch css data
+    const fontCssData: string[][] = await Promise.all(
+      metadata.weights.map((weight) => {
+        fontCssNames.push([weight, []]);
+        return Promise.all(
```

Comment:
Is Promise.all necessary in this section? At least, I don't think we're returning any promises within the function.

Reply:
`Promise.all` is necessary to fetch the CSS data in parallel, because its looping through all the weights, and inside of that, all the styles, then it creates a `Promise.all` for all with weights styles to finish fetching. Then theres the main `Promise.all` for when all the `Promise.all`'s for the different weight fetching is done. This results in a promise for when all of the css files are done downloading and allows them to fetch in parallel (all at the same time).